### ISSUE TITLE
Cant get specific metadata from listFolderWithPath results (SwiftyDropboxObjC).

### ISSUE THREAD

**Issue by wyllys66** [dropbox/SwiftyDropbox#399]
Im using the latest SwiftyDropboxObjC(10.0.1) and trying to list a folder and then view the file and sub-folder metadata in an iOS app.  This all worked fine with the old objc-v2 cocoapod, but I am trying to keep up with the latest SPM packaging and APIs, but ran into a snag.

The calls to "isKindOfClass:...] always fail, it seems that the entries returned are just generic "DBXFilesMetadata" and it is not possible to determine if they are files or folders or get extra data such as the "clientModified" field or file size.

Is there some trick to getting this to work in an Obj-C app?

In my case the "/" level folder being listed returns about 11 items, some of which are actual files and others are sub-folders.

This is the code more or less (left out some error handling logic, etc):

```

	[[dbc.files listFolderWithPath:@"/"] responseWithCompletionHandler:^(DBXFilesListFolderResult * _Nullable result,
                                                                           DBXFilesListFolderError * _Nullable routeError,
                                                                           DBXCallError * _Nullable error) {
	    for (DBXFilesMetadata *entry in result.entries) {
		if ([entry isKindOfClass:[DBXFilesFileMetadata class]]) {
			DBXFilesFileMetadata *fileMetadata = (DBXFilesFileMetadata *)entry;
			NSLog(@"File: %@\n", fileMetadata);
			
		} else if ([entry isKindOfClass:[DBXFilesFolderMetadata class]]) {
			DBXFilesFolderMetadata *folderMetadata = (DBXFilesFolderMetadata *)entry;
			NSLog(@"Folder: %@\n", folderMetadata);
		}
            }
        }];
```

**Comment by greg-db** [comment]
Thanks for writing this up! We'll check on this and follow up here once we have some information on this.

**Comment by julianlocke** [comment]
@wyllys66 this issue is fixed in version 10.0.2. Thanks for raising it and let us know if you experience any other oddities.
